routing/router.go

```diff
@@ -92,3 +108,52 @@ func metricHandler(next http.Handler, route Route) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+// serveEnvJsFile generates the env.js file needed by the UI from Kiali configs. The
+// generated file is sent to the HTTP response.
+func serveEnvJsFile(w http.ResponseWriter) {
+	conf := config.Get()
+	var body string
+	if len(conf.Server.WebHistoryMode) > 0 {
+		body += fmt.Sprintf("window.HISTORY_MODE='%s';", conf.Server.WebHistoryMode)
+	}
+
+	if webRoot := strings.TrimSuffix(config.Get().Server.WebRoot, "/"); len(webRoot) > 0 {
+		body += fmt.Sprintf("window.WEB_ROOT='%s';", webRoot)
+	}
+
+	w.Header().Set("content-type", "text/javascript")
+	_, err := io.WriteString(w, body)
+	if err != nil {
+		log.Errorf("HTTP I/O error [%v]", err.Error())
+	}
+}
+
+// serveIndexFile takes UI's index.html as a template to generate a modified index file that takes
+// into account the web_root path configured in the Kiali CR. The result is sent to the HTTP response.
+func serveIndexFile(w http.ResponseWriter) {
+	webRootPath := config.Get().Server.WebRoot
+	webRootPath = strings.TrimSuffix(webRootPath, "/")
+
+	path, _ := filepath.Abs("./console/index.html")
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Errorf("File I/O error [%v]", err.Error())
+		return
```

Comment:
Shouldn't we return some kind of HTTP error status code and message here? It seems like a simple return will not provide the client any indication that a problem occurred. We should return some kind of 500 error here I think (writing it to "w")

Reply:
I added the message.
However, it broke some tests. I fixed them, but I don't like the fixes.